### PR TITLE
Contributing to Cortex XSOAR Content

### PR DESCRIPTION
**Pull request by dmelgar-ssc**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: - Added fixes requested in this PR https://github.com/demisto/content/pull/32230

## Description
The update includes three new playbooks and a few new commands to power them. This new content is able to interact with SecurityScorecard's Event Log, pulling in event data and then fetching detailed information for particularly relevant events. Furthermore it is able to bring in metadata for issues such as description and resolution information.

### CODE REVIEW

**Comment by kgal-pan** [comment]
@dmelgar-ssc - Is there a reason we need to open this PR to merge into https://github.com/demisto/content/pull/32230 or can we continue working on that PR? You should be able to run:

```bash
gh pr checkout 32230
```

To pull the work into your local environment.

**Comment by dmelgar-ssc** [comment]
The previous pull request (32230) is based on a fork created by a contributor who is no longer working in Security Scorecard and the fork was created using his account.
That's why I've created a new one with a new fork using my account, the idea is to apply the requested changes on the original PR from this pull request having all track of it.
Also, I'm taking this work from an internal ticket in Security Scorecard that we tracking these changes.
After merging the Pull Requests, I will register myself as a contributor and create my pull requests directly.

File: Packs/SecurityScorecard/Integrations/SecurityScorecard/SecurityScorecard.yml [hunk 1]
```diff
@@ -53,31 +52,35 @@ configuration:
   display: Use system proxy settings
   type: 8
   required: false
+- display: Portfolio ID
```
Comment (YuvHayun) [inline Packs/SecurityScorecard/Integrations/SecurityScorecard/SecurityScorecard.yml:55]: I don't seem to find this param in the code, where are you using it?

File: Packs/SecurityScorecard/Integrations/SecurityScorecard/SecurityScorecard.yml [hunk 2]
```diff
@@ -340,51 +343,107 @@ script:
...
-      description: Client domain. This value is identical to the input of the domain argument
+      description: Client domain. This value is identical to the input of the domain argument.
       type: String
     - contextPath: SecurityScorecard.Service.categories
-      description: Vendor service provider, e.g. mail_provider, nameserver_provider
+      description: Vendor service provider, e.g. mail_provider, nameserver_provider.
       type: array
+  - name: securityscorecard-company-events-get
+    arguments:
+    - name: domain
```
Comment (YuvHayun) [inline Packs/SecurityScorecard/Integrations/SecurityScorecard/SecurityScorecard.yml:396]: Let's add description.

File: Packs/SecurityScorecard/Integrations/SecurityScorecard/SecurityScorecard.yml [hunk 3]
```diff
@@ -340,51 +343,107 @@ script:
...
+      required: true
+      type: textArea
+    - name: date_from
+    - name: date_to
```
Comment (YuvHayun) [inline Packs/SecurityScorecard/Integrations/SecurityScorecard/SecurityScorecard.yml:400]: same here.

File: Packs/SecurityScorecard/Integrations/SecurityScorecard/SecurityScorecard.yml [hunk 4]
```diff
@@ -340,51 +343,107 @@ script:
...
+    outputs:
+    - contextPath: SecurityScorecard.Events.Event.ssc_event_id
```
Comment (YuvHayun) [inline Packs/SecurityScorecard/Integrations/SecurityScorecard/SecurityScorecard.yml:402]: Make sure to add a description field for all outputs, arguments & commands.

File: Packs/SecurityScorecard/Integrations/SecurityScorecard/SecurityScorecard.yml [hunk 5]
```diff
@@ -340,51 +343,107 @@ script:
...
+      type: number
+    - contextPath: SecurityScorecard.Events.Event.issue_type
+      type: string
+    - contextPath: SecurityScorecard.Events.Event.severity
+      type: string
+    - contextPath: SecurityScorecard.Events.Event.factor
+      type: string
+    - contextPath: SecurityScorecard.Events.Event.ssc_detail_url
+      type: string
+  - name: securityscorecard-company-findings-get
```
Comment (YuvHayun) [inline Packs/SecurityScorecard/Integrations/SecurityScorecard/SecurityScorecard.yml:419]: No outputs for this command?

**Review by YuvHayun — CHANGES_REQUESTED** [review]
Hey @dmelgar-ssc, thx for your contribution!
Looks good in general, please see my notes.
Also, please make sure to add tests for the commands you've added.

**Comment by YuvHayun** [comment]
@melamedbn @MLainer1 ready for your review.

**Comment by melamedbn** [comment]
Hi @dmelgar-ssc,

I've taken some time to review the SecurityScorecard playbooks and have a few suggestions that could enhance their functionality and clarity:

### SecurityScorecard Event
- **Playbook Description**: It's important to add a description for clarity and understanding of the playbook's purpose.
- **Naming**: Could we find a more descriptive term than 'event' that better encapsulates the playbook's functionality?
- **Playbook Inputs**: Please provide descriptions and default values for the playbook's inputs for better user guidance.
- **Task Clarification (Tasks 2, 4, 6)**: These tasks reference playbook inputs that don't seem to exist. Could you please recheck and rectify this?
- **Playbook Purpose**: The current playbook appears to create an incident from incident info, potentially leading to duplicated incidents. This process might be more efficiently handled with a classifier. Could you elaborate on the playbook's intended use?

### SecurityScorecard Events
- **Playbook Description**: It's important to add a description for clarity and understanding of the playbook's purpose.
- **Naming**: Could we find a more descriptive term than 'event' that better encapsulates the playbook's functionality?
- **Hardcoded Values in Inputs**: Are there reasons for the hardcoded values in the playbook inputs? It's generally advisable to use adjustable inputs.
- **Playbook Outputs**: Have you considered whether this playbook requires outputs?

### SecurityScorecard Sync
- **Playbook Description**: It's important to add a description for clarity and understanding of the playbook's purpose.
- **Naming**: Could we find a more descriptive term than 'event' that better encapsulates the playbook's functionality?
- **Task 5 - 'portfolio_id' Input**: This task is missing the mandatory input 'portfolio_id'. Please include it.
- **Avoiding Hardcoded Values**: Adjustable playbook inputs are recommended over hardcoded values for flexibility and wider applicability.

### General Note
- **Combining Playbooks**: It seems feasible and efficient to combine all three playbooks into one, given their overlapping functionality and scope.

Looking forward to your thoughts on these suggestions.

Best regards,

Ben

**Comment by dmelgar-ssc** [comment]
> Hey @dmelgar-ssc how is it going? Let me know if you need help with my comments / @melamedbn comments.

Hey @melamedbn, I had an issue trying to complete the contributor form (getting the required ID from slack) I received the invite but it was canceled immediately. 
I am working with our Product Manager to get the information required to complete the release notes and the information requested on descriptions fields and others.

**Comment by YuvHayun** [comment]
Alright, let me know if you need any help from my side @dmelgar-ssc

**Comment by YuvHayun** [comment]
Closing the pr for now.
Feel free to reopen when you get to address the comments.

**Pull request closed without merging**